### PR TITLE
Prepare v3.6 release. 

### PR DESCRIPTION
**Pull request by carltongibson**
Hey @andrewgodwin. 

I'm looking at putting out a v3.6 that we can use for Django 4.2a1. 

I looked through the other PRs. I think they're all in various stages of waiting except #337. 

- [x] Added release note for coroutine detection shims. (Here.) 
- [x] Merge #337
- [x] Bump version, tag, and release. 

Do you think this is right? 
Can you think of anything else you want to add? 

Thanks.

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,5 +1,20 @@
-UNRELEASED
-----------
+3.6.0 (UNRELEASED)
```
Comment (carltongibson) [inline CHANGELOG.txt:1]: Date needed before actual release.

**Comment by carltongibson** [comment]
Hey @andrewgodwin — thanks! 

I've rebased. Point 3 is ready to go. I just checked, I have PyPI perms, so I can release (likely Tuesday) unless you'd like to do the honours. 

🚀

File: CHANGELOG.txt [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+------------------
+
+* Two new functions are added to the ``asgiref.sync`` module: ``iscoroutinefunction()``
+  and ``markcoroutinefunction()``.
+
+  Python 3.12 deprecates asyncio.iscoroutinefunction() as an alias for
```
Comment (smithdc1) [inline CHANGELOG.txt:7]:
```suggestion
  Python 3.12 deprecates ``asyncio.iscoroutinefunction()`` as an alias for
```

File: CHANGELOG.txt [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+  inspect.iscoroutinefunction(), whilst also removing the _is_coroutine marker.
```
Comment (smithdc1) [inline CHANGELOG.txt:8]:
```suggestion
  ``inspect.iscoroutinefunction()``, whilst also removing the ``_is_coroutine`` marker.
```

Should these two be in backticks?

Similar below too.


Side comment.  Whilst. 😅👍

File: CHANGELOG.txt [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+  The latter is replaced with the inspect.markcoroutinefunction decorator.
+
+  The new ``asgiref.sync`` are compatibility shims for these functions that can be used
```
Comment (smithdc1) [inline CHANGELOG.txt:11]:
```suggestion
  The new ``asgiref.sync`` functions are compatibility shims for these functions that can be used
```

Seems there's a word missing here? Not sure if functions is quite right

**Comment by carltongibson** [comment]
Thanks @smithdc1! 🎩

**Comment by carltongibson** [comment]
OK, no-one shouted — I'm going to take that as 🟢 🙂

**Comment by carltongibson** [comment]
https://pypi.org/project/asgiref/3.6.0/

**Pull request merged**